test/runtests.jl

```diff
@@ -67,18 +176,84 @@ end
 end
 
 @testset "LinQuant along dimension" begin
-    A = rand(Float32,10,20,30,40)
-    Q = LinQuant32Array(A,4)
-    @test A ≈ Array{Float32}(Q)
+    @testset "Default extrema" begin
+        A = rand(Float32,10,20,30,40)
+        Q = LinQuant32Array(A,4)
+        @test A ≈ Array{Float32}(Q)
 
-    Q = LinQuant24Array(A,4)
-    @test A ≈ Array{Float32}(Q)
+        Q = LinQuant24Array(A,4)
+        @test A ≈ Array{Float32}(Q)
 
-    Q = LinQuant16Array(A,4)
-    @test A ≈ Array{Float32}(Q)
+        Q = LinQuant16Array(A,4)
+        @test A ≈ Array{Float32}(Q)
 
-    Q = LinQuant8Array(A,4)
-    @test all(isapprox.(A,Array{Float32}(Q),atol=1e-1))
+        Q = LinQuant8Array(A,4)
+        @test all(isapprox.(A,Array{Float32}(Q),atol=1e-1))
+
+        A = rand(Float32,10,20,30,40)
+        Q = LinQuantArray{Int32}(A,4)
+        @test A ≈ Array{Float32}(Q)
+
+        Q = LinQuantArray{Int24}(A,4)
+        @test A ≈ Array{Float32}(Q)
+
+        Q = LinQuantArray{Int16}(A,4)
+        @test A ≈ Array{Float32}(Q)
+
+        Q = LinQuantArray{Int8}(A,4)
+        @test all(isapprox.(A,Array{Float32}(Q),atol=1e-1))
+    end 
+
+    @testset "Custom extrema" begin
+        for ext in [(0.1, 0.6), (0.22, 0.75), (0.3, 0.5)]
+            A = rand(Float32,10,20,30,40)
+
+            A2 = Array{Float32}(LinQuantArray{UInt32}(A, 4; extrema=ext))
+            @test A2 == Array{Float32}(LinQuantArray{UInt32}(A2, 4; extrema=ext))
+
+
+            A2 = Array{Float32}(LinQuantArray{UInt24}(A, 4; extrema=ext))
+            @test A2 == Array{Float32}(LinQuantArray{UInt24}(A2, 4; extrema=ext))
+            
+            A2 = Array{Float32}(LinQuantArray{UInt16}(A, 4; extrema=ext))
+            @test A2 == Array{Float32}(LinQuantArray{UInt16}(A2, 4; extrema=ext))
+
+            A2 = Array{Float32}(LinQuantArray{UInt8}(A, 4; extrema=ext))
+            all(isapprox.(A2, Array{Float32}(LinQuantArray{UInt8}(A2, 4; extrema=ext)), atol=1e-1))
+
+            A2 = Array{Float32}(LinQuantArray{Int32}(A, 4; extrema=ext))
+            @test A2 == Array{Float32}(LinQuantArray{Int32}(A2, 4; extrema=ext))
+
+            A2 = Array{Float32}(LinQuantArray{Int24}(A, 4; extrema=ext))
+            @test A2 == Array{Float32}(LinQuantArray{Int24}(A2, 4; extrema=ext))
+            
+            A2 = Array{Float32}(LinQuantArray{Int16}(A, 4; extrema=ext))
+            @test A2 == Array{Float32}(LinQuantArray{Int16}(A2, 4; extrema=ext))
+
+            A2 = Array{Float32}(LinQuantArray{Int8}(A, 4; extrema=ext))
+            all(isapprox.(A2, Array{Float32}(LinQuantArray{Int8}(A2, 4; extrema=ext)), atol=1e-1))
+
+        end
+    end
+
+    @testset "Negative values" begin
+
+        A = 2 * rand(Float32,10,20,30,40) .- 1
+        @test minimum(A) < 0
+        @test maximum(A) > 0
+
+        Q = LinQuantArray{Int32}(A, 4)
+        @test A ≈ Array{Float32}(Q)
```

Comment:
if you test after the roundtrip then you can actually test for equality because rounding is an idempotent operation, introducing the rounding error only the first time

Reply:
Actually I just copied your previous tests. I thought you might have a reason to so.

Reply:
Just added idempotency tests